jung-api/src/main/java/edu/uci/ics/jung/graph/DelegateCTree.java

```diff
@@ -206,13 +211,12 @@ public boolean removeNode(N node) {
     if (!nodes().contains(node)) {
       return false;
     }
+    // TODO: Replace `Graphs.reachableNodes` with `Traverser.forTree.breadthFirst` for lazy eval?
     for (N nodeToRemove : Graphs.reachableNodes(delegate, node)) {
       delegate.removeNode(nodeToRemove);
       depths.remove(nodeToRemove);
     }
-    if (root.isPresent() && root.get().equals(node)) {
-      setRoot(Optional.empty());
-    }
+    root.filter(value -> value.equals(node)).ifPresent(value -> setRoot(Optional.empty()));
```

Comment:
Hmm, with hindsight, although this is more functional, I think it's actually harder to read than before.

If I remember to do so, I'll revert this particular line.

Reply:
This is a reminder to please do revert this specific change, because I agree that it's harder to read this way.  :)

Reply:
Done!